ttnn/cpp/ttnn/tensor/tensor_layout.cpp

```diff
@@ -0,0 +1,358 @@
+      mAlignment(alignment) {
+
+    mLayout = mTileSize.height() == 1 ? Layout::ROW_MAJOR : Layout::TILE;
+    initializeAlignment();
+    validateCustomAlignment();
+}
+
+TensorLayout::TensorLayout(DataType dataType, Layout layout, const MemoryConfig& memoryConfig, const Alignment& alignment)
+    : mLayout(layout),
+      mDataType(dataType),
+      mMemoryConfig(memoryConfig),
+      mAlignment(alignment) {
+
+    if (mLayout == Layout::TILE) {
+        mTileSize = Size(32, 32);
+    }
+    else {
+        mTileSize = Size(1, 1); // width 1 makes no sense? need juggestions
+    }
+
+    initializeAlignment();
+    validateCustomAlignment();
+}
+
+namespace {
+
+// This function is used to convert the legacy padded shape to Alignment
+// This is a temporary solution until we get rid of the LegacyPaddedShape
+// Keep in mind:
+//    The benefit of Alignment compared to LegacyPaddedShape is that it allows
+//    to pad between Channels and Batches with a row granularity.
+// Examples:
+// 1. Given LegacyPaddedShape (1, 1, 16, 20), alignment is (1, 1, 16, 20)
+//    means Shape (1, 1, 16, 16) which in theory takes physical space (1 * 1 * 16, 16)
+//    will take (1 * 1 * 16, 20)
+//
+// 2. Given LegacyPaddedShape (1, 1, 32, 32), alignment is (1, 1, 32, 32)
+//    Shape (1, 1, 16, 16)
+//      which in theory takes physical space (1 * 1 * 16, 16)
+//      will take (1 * 1 * 32, 32)
+//
+// 3. Given LegacyPaddedShape (2, 3, 32, 32), alignment is (2, 3*32, 32, 32).
+//    Shape (2, 2, 16, 16)
+//      which in theory takes physical space (2 * 2 * 16, 16)
+//      will take (2 * 3 * 32, 32)
+
+Alignment legacyPaddedShapeToAlignment(const ttnn::SimpleShape& legacyPaddedShape) {
+    std::vector<uint32_t> values;
+    const auto rank = legacyPaddedShape.rank();
+    values.resize(rank);
+
+    if(rank > 1) {
+        values[rank - 1] = legacyPaddedShape[rank - 1];
+    }
+    if(rank > 2) {
+        values[rank - 2] = legacyPaddedShape[rank - 2];
+    }
+    for (size_t i = rank - 3; i >= 0; i--) {
+        values[i] = legacyPaddedShape[i] * values[i + 1];
+    }
+
+    Alignment result(values);
+    return result;
+}
+}
+
+// Private constructor to create TensorLayout from LegacyPaddedShape
+TensorLayout::TensorLayout(DataType dataType, Layout layout, const MemoryConfig& memoryConfig, const ttnn::SimpleShape& legacyPaddedShape)
+    : TensorLayout(dataType, layout, memoryConfig, legacyPaddedShapeToAlignment(legacyPaddedShape)) {
+    mLegacyPaddedShape = legacyPaddedShape;
+}
+
+TensorLayout TensorLayout::fromLegacyPaddeShape(DataType dataType, Layout layout, const MemoryConfig& memoryConfig, const ttnn::SimpleShape& legacyPaddedShape) {
+    return TensorLayout(dataType, layout, memoryConfig, legacyPaddedShape);
+}
+
+void TensorLayout::initializeAlignment() {
+    if(mAlignment.size() != 0)
+        return;
+
+    switch(mLayout) {
+        case Layout::ROW_MAJOR: {
+            mAlignment = Alignment({sizeof(uint32_t) % element_size_bytes()});
```

Comment:
I believe this should be `sizeof(uint32_t) / element_size_bytes()` instead of `sizeof(uint32_t) % element_size_bytes()`

Reply:
yes, found during testing, thank you for the catch